src/Build/Logging/BinaryLogger/BinaryLogReplayEventSource.cs

```diff
@@ -19,7 +19,7 @@ public sealed class BinaryLogReplayEventSource : EventArgsDispatcher
         /// <param name="sourceFilePath">The full file path of the binary log file</param>
         public void Replay(string sourceFilePath)
         {
-            using (var stream = new FileStream(sourceFilePath, FileMode.Open))
+            using (var stream = new FileStream(sourceFilePath, FileMode.Open, FileAccess.Read))
```

Comment:
Should we make FileShare explicit as well while we're changing this?

Reply:
Would you suggest just adding `FileShare.Read` there?

Reply:
That's the default, and it seems sensible to me.